### DISCUSSION TITLE
In rule configuration, why is the PromQL lost after switching between the Metric and Host tabs?

### DISCUSSION THREAD

**Discussion by czxin788** [ccfos/nightingale#1897] [General]
请问规则配置里，切换Metric和Host两个标签后，metric里面的promql里面的内容没了是怎么回事：
![image](https://github.com/ccfos/nightingale/assets/13861904/6ab58ced-c767-45ba-8711-fa926cd5a7f8)

![image](https://github.com/ccfos/nightingale/assets/13861904/ca10e506-8b22-420a-a8ee-e0a7b179e2c3)
![image](https://github.com/ccfos/nightingale/assets/13861904/e9d175ed-1398-4c4f-884d-86a76836851b)

**Comment by UlricQin** [comment]
@jsers 这个可以稍微优化一下。再尚未提交之前，前端不清空，只有提交的那一刻，只保留特定规则相关的配置项，其他的丢弃即可
